test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpect100Test.java

```diff
@@ -230,6 +218,10 @@ private void handle_connection(BufferedReader in, PrintStream out)
             } else {
                 defaultResponse(out);
             }
+            // wait until the client closes the socket
+            while (line != null) {
```

Comment:
Hello Daniel, does the code ever enter this `while` loop? If I'm reading the above few lines correctly, then when the code exits the previous `do while` loop then `line` would be `null` and `line` appears to continue being `null` when it reaches here.

Reply:
Hi Jaikiran, the previous write loop is usually exited with a zero-length line. Null is only returned on EOF.

Reply:
Thank you that explains it.